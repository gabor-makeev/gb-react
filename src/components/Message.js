export const Message = (props) => {
  return <p>Message: "{props.content}"</p>
}