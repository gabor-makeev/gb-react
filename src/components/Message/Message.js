import style from './Message.module.scss'

export const Message = (props) => {

  const copyMessageContent = () => {
    navigator.clipboard.writeText(props.content)
      .then(() => {
        const messageBox = document.querySelector(`.${style.message}`)
        const messageOriginalValue = messageBox.textContent

        messageBox.textContent = 'text copied'

        setTimeout(() => {
          messageBox.textContent = messageOriginalValue
        }, '1500')
      })
  }

  return <p className={style.message} onClick={copyMessageContent}>Message: "{props.content}"</p>
}