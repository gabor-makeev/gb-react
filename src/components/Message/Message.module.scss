.message {
  background-color: #525252;
  color: white;
  padding: 30px;
  border-radius: 30px;
  margin: 0;
  transition: 0.3s;

  &:hover {
    background-color: #777777;
    box-shadow: rgba(0, 0, 0, 0.35) 0 5px 15px;
  }
}