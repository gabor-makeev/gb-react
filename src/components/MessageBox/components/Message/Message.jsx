import style from './Message.module.scss';
import ListItem from '@mui/material/ListItem';
import { AUTHORS, STYLES } from '../../../../constants';

export const Message = ({
  message,
  padding = '15px 15px',
  maxWidth = '275px',
}) => {
  const messageStyle = {
    backgroundColor:
      message.author === AUTHORS.bot
        ? STYLES.color.system
        : STYLES.color.secondary,
    maxWidth: maxWidth,
    padding: padding,
    boxSizing: 'border-box',
    position: 'relative',
    justifyContent: 'flex-end',
    borderRadius: '20px',
    listStyleType: 'none',
    fontFamily: 'Arial, sans-serif',
    fontSize: '14px',
  };

  return (
    <ListItem sx={messageStyle} className={style.message}>
      {message.text}
      <span className={style['message__author-sign']}>{message.author}</span>
    </ListItem>
  );
};
