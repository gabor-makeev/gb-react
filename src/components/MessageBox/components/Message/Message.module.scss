.message {
  //Custom styles should go here
  &__author-sign {
    position: absolute;
    left: 15px;
    top: 5px;
    font-size: 12px;
  }
}