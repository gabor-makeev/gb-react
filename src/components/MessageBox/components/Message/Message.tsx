import style from './Message.module.scss';
import ListItem from '@mui/material/ListItem';
import { AUTHORS, STYLES } from '../../../../constants';
import { FC } from 'react';
import { MessageItem } from 'src/default-types';

interface MessageProps {
  message: MessageItem;
  padding?: string;
  maxWidth?: string;
}

export const Message: FC<MessageProps> = ({
  message,
  padding = '15px 15px',
  maxWidth = '275px',
}) => {
  const messageStyle = {
    backgroundColor:
      message.author === AUTHORS.bot
        ? STYLES.color.system
        : STYLES.color.secondary,
    maxWidth: maxWidth,
    padding: padding,
    boxSizing: 'border-box',
    position: 'relative',
    justifyContent: 'flex-end',
    borderRadius: '20px',
    listStyleType: 'none',
    fontFamily: 'Arial, sans-serif',
    fontSize: '14px',
  };

  return (
    <ListItem sx={messageStyle} className={style.message}>
      {message.text}
      <span className={style['message__author-sign']}>{message.author}</span>
    </ListItem>
  );
};
