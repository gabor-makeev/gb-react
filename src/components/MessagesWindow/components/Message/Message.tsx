import style from './Message.module.scss';
import { AUTHORS } from '../../../../constants';
import { FC } from 'react';
import { MessageItem } from 'src/default-types';
import { MessageListItem } from '../../../StyledMUIComponents/MessageListItem';
import classNames from 'classnames';

interface MessageProps {
  message: MessageItem;
  variant?: string;
}

export const Message: FC<MessageProps> = ({ message, variant = 'medium' }) => {
  let MessageListItemClasses = classNames(style[`message__type-${variant}`]);

  if (message.author === AUTHORS.bot) {
    MessageListItemClasses += ` ${style['message__system-background']}`;
  } else {
    MessageListItemClasses += ` ${style['message__user-background']}`;
  }

  return (
    <MessageListItem className={MessageListItemClasses} disablePadding={true}>
      {message.text}
      <span className={style['message__author-sign']}>{message.author}</span>
    </MessageListItem>
  );
};
