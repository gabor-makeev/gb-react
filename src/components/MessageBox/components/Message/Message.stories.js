import { Message } from './Message';
import { DUMMY_CONTENT } from '../../../../constants';

export default {
  title: 'Message',
  component: Message,
};

const dummyMessage = DUMMY_CONTENT.messages[0];

const Template = (args) => <Message message={dummyMessage} {...args} />;

export const Primary = Template.bind({});

export const Big = Template.bind({});
Big.args = {
  padding: '25px 15px',
  maxWidth: '300px',
};

export const Small = Template.bind({});
Small.args = {
  padding: '15px 10px',
  maxWidth: '250px',
};
