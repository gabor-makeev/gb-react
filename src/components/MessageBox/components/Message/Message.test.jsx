import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Message } from './Message';
import { DUMMY_CONTENT, STYLES } from '../../../../constants';

const dummyMessage = DUMMY_CONTENT.messages[0];

describe('Message', () => {
  it('should render', () => {
    render(<Message message={dummyMessage} />);
  });

  it('should render text in <li>-tag', () => {
    render(<Message message={dummyMessage} />);
    expect(screen.getByText(dummyMessage.text)).toBeInTheDocument();
  });

  it('should render text in <span>-tag', () => {
    render(<Message message={dummyMessage} />);
    expect(screen.getByText(dummyMessage.author)).toBeInTheDocument();
  });

  it('should use default messageStyle property when such not passed', () => {
    render(<Message message={dummyMessage} />);
    expect(screen.getByText(dummyMessage.text)).toHaveStyle(
      `background-color: ${STYLES.color.secondary}`
    );
  });

  it('should use style passed in messageStyle prop', () => {
    render(
      <Message
        message={dummyMessage}
        messageStyle={{ backgroundColor: 'red' }}
      />
    );

    expect(screen.getByText(dummyMessage.text)).toHaveStyle(
      'background-color: red'
    );
  });

  it('should render with snapshot', () => {
    const { asFragment } = render(<Message message={dummyMessage} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
