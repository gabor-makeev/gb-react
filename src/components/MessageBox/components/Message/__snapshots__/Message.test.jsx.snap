// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render with snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding message css-1s94d1u-MuiListItem-root"
  >
    placerat duis ultricies lacus sed turpis tincidunt id aliquet risus
    <span
      class="message__author-sign"
    >
      Gabor
    </span>
  </li>
</DocumentFragment>
`;
